src/main/java/org/opentripplanner/middleware/tripMonitor/jobs/CheckMonitoredTrip.java

```diff
@@ -46,29 +43,49 @@
     private static final Logger LOG = LoggerFactory.getLogger(CheckMonitoredTrip.class);
 
     public final MonitoredTrip trip;
+
     /**
      * Used to track the various check trip notifications and construct email/SMS messages.
      */
     public final Set<TripMonitorNotification> notifications = new HashSet<>();
+
     /**
-     * The index of the matching {@link Itinerary} as found in the OTP {@link OtpResponse} planned as part of this check.
+     * The matching itinerary from the previous run of this job.
      */
-    public Itinerary matchingItinerary;
+    public Itinerary previousMatchingItinerary;
+
     /**
-     * The OTP response planned to check the stored {@link Itinerary} against.
+     * The matching {@link Itinerary} that is used during this check. It is calculated in the following ways:
+     * - the trip's Itinerary if this check is being ran for the first time.
+     * - the trip's JourneyState if the check has already been performed
+     * - an OTP response if the trip is currently active
      */
-    public OtpResponse otpResponse;
+    public Itinerary matchingItinerary;
+
     /**
      * Tracks the time the notification was sent to the user.
      */
     public long notificationTimestampMillis = -1;
 
+    /**
+     * The journey state that was calculated from the previous run of this job.
+     */
+    public JourneyState previousJourneyState;
+
+    /**
+     * An updated journey state that will be saved to the MonitoredTrip record after this check.
+     */
+    public JourneyState journeyState;
+
     public String targetDate;
 
-    private ZonedDateTime targetZonedDateTime;
+    ZonedDateTime targetZonedDateTime;
```

Comment:
Are both `targetDate` and `targetZonedDateTime` necessary? Might be helpful to have some javadoc explaining what they're used for.

Reply:
Huh, yeah, `targetDate` isn't necessary as a field in this class. Removed in 1987213b6754ffa98f4748820b628be64f9443f4.